src/game_interpreter.cpp

```diff
@@ -2422,125 +2551,108 @@ bool Game_Interpreter::CommandPlayMemorizedBGM(RPG::EventCommand const& /* com *
+		return 1;
 	}
-
-	bool time = false;
-	int time_id = 0;
-
-	bool check_decision = com.parameters[3] != 0;
-	bool check_cancel = com.parameters[4] != 0;
-	bool check_numbers = false;
-	bool check_arith = false;
-	bool check_shift = false;
-	bool check_down = false;
-	bool check_left = false;
-	bool check_right = false;
-	bool check_up = false;
-	int result = 0;
-	size_t param_size = com.parameters.size();
-
-	// Use a function pointer to check triggered keys if it waits for input and pressed keys otherwise
-	bool(*check)(Input::InputButton) = wait ? Input::IsTriggered : Input::IsPressed;
-
-	if (param_size < 6) {
-		// For Rpg2k <1.50
-		bool check_dir = com.parameters[2] != 0;
-		check_down = check_dir;
-		check_left = check_dir;
-		check_right = check_dir;
-		check_up = check_dir;
-	} else if (param_size < 11) {
-		// For Rpg2k >=1.50
-		check_shift = com.parameters[5] != 0;
-		check_down = param_size > 6 ? com.parameters[6] != 0 : false;
-		check_left = param_size > 7 ? com.parameters[7] != 0 : false;
-		check_right = param_size > 8 ? com.parameters[8] != 0 : false;
-		check_up = param_size > 9 ? com.parameters[9] != 0 : false;
-	} else {
-		// For Rpg2k3
-		check_numbers = com.parameters[5] != 0;
-		check_arith = com.parameters[6] != 0;
-		time_id = com.parameters[7];
-		time = com.parameters[8] != 0;
-		check_shift = com.parameters[9] != 0;
-		check_down = com.parameters[10] != 0;
-		check_left = param_size > 11 ? com.parameters[11] != 0 : false;
-		check_right = param_size > 12 ? com.parameters[12] != 0 : false;
-		check_up = param_size > 13 ? com.parameters[13] != 0 : false;
-	}
-
-	if (check_down && check(Input::DOWN)) {
-		result = 1;
+	if (keys[Keys::eLeft] && check(Input::LEFT)) {
+		return 2;
 	}
-	if (check_left && check(Input::LEFT)) {
-		result = 2;
+	if (keys[Keys::eRight] && check(Input::RIGHT)) {
+		return 3;
 	}
-	if (check_right && check(Input::RIGHT)) {
-		result = 3;
+	if (keys[Keys::eUp] && check(Input::UP)) {
+		return 4;
 	}
-	if (check_up && check(Input::UP)) {
-		result = 4;
+	if (keys[Keys::eDecision] && check(Input::DECISION)) {
+		return 5;
 	}
-	if (check_decision && check(Input::DECISION)) {
-		result = 5;
+	if (keys[Keys::eCancel] && check(Input::CANCEL)) {
+		return 6;
 	}
-	if (check_cancel && check(Input::CANCEL)) {
-		result = 6;
+	if (keys[Keys::eShift] && check(Input::SHIFT)) {
+		return 7;
 	}
-	if (check_shift && check(Input::SHIFT)) {
-		result = 7;
-	}
-	if (check_numbers) {
+	if (keys[Keys::eNumbers]) {
 		for (int i = 0; i < 10; ++i) {
 			if (check((Input::InputButton)(Input::N0 + i))) {
-				result = 10 + i;
+				return 10 + i;
 			}
 		}
 	}
-	if (check_arith) {
+	if (keys[Keys::eOperators]) {
 		for (int i = 0; i < 5; ++i) {
 			if (check((Input::InputButton)(Input::PLUS + i))) {
-				result = 20 + i;
+				return 20 + i;
 			}
 		}
 	}
 
-	if (var_id > 0) {
-		Game_Variables.Set(var_id, result);
-		Game_Map::SetNeedRefresh(Game_Map::Refresh_Map);
-	}
+	return 0;
```

Comment:
> Refactor keyinputproc to use LSD chunks properly 

In the old code when holding e.g. Shift + Num0 the result would be Num0, in the new code it would be shift. Is this correct?

Reply:
yeah this change is wrong, when I just create an event which changes the graphic based on Confirm or Shift in RPG\_RT it shows shift (7) when holding both, in Player is now confirm (5)

Reply:
For multiple keys pressed, the logic seems to be greatest key value wins. I totally overlooked this situation. This was a really great find!